core/src/main/java/org/apache/carbondata/core/scan/filter/executer/RowLevelRangeGrtThanFiterExecuterImpl.java

```diff
@@ -366,7 +364,7 @@ private BitSet setFilterdIndexToBitSet(DimensionColumnDataChunk dimensionColumnD
     BitSet bitSet = new BitSet(numerOfRows);
     byte[][] filterValues = this.filterRangeValues;
     // binary search can only be applied if column is sorted
-    if (isNaturalSorted) {
+    if (isNaturalSorted && !dimensionColumnDataChunk.isNoDicitionaryColumn()) {
```

Comment:
why this check is required? even no dictionary columns also can be sorted right?

Reply:
In this PR, plain value (no dictionary) is using adaptive encoding, it is not sorted. I think it will make store size smaller but for some filter query it will be slower. Any suggestion?